contracts/VaultV3.vy

```diff
@@ -650,10 +620,20 @@ def _process_report(strategy: address) -> (uint256, uint256):
         self.strategies[strategy].total_gain += gain
         # update current debt after processing management fee
         self.strategies[strategy].current_debt += gain
-        self.locked_profit = self._calculate_locked_profit() + gain - total_fees
 
+        gain_without_fees: uint256 = gain - total_fees
+        if PROFIT_MAX_UNLOCK_TIME == 0:
+            self.total_debt_ += gain_without_fees  + unlocked_profit
+        else:
+            # The new locking period is the weighted average between the remaining time and the profit_max_unlock_time. 
+            # The weight used is the profit (pending_profit vs new_profit)
+            new_profit_locking_period: uint256 = (pending_profit * remaining_time + gain_without_fees * PROFIT_MAX_UNLOCK_TIME) / (pending_profit + gain_without_fees)
```

Comment:
may be misunderstanding but it seems to me theres a path where pending_profit can be 0 here which will lead to a revert on division?

Pencil Test:

self.profit_distribution_rate_ != 0
block.timestamp > profit_end_date
gain > 0
PROFIT_MAX_UNLOCK_TIME != 0

Is this scenario possible?

Reply:
There is a maximum fee cap: https://github.com/jmonteer/yearn-vaults-v3/blob/master/contracts/periphery/FeeManager.vy#L86-L90

However, if the `MAX_SHARE` value were to be 100%, the scenario you presented could happen.